### PR TITLE
Dockerfile optimisations (docker image size)

### PR DESCRIPTION
**Pull request by zwindler**
- Clear apt to gain a few MB (50 MB)
- Use --no-install-recommends (200 MB)
- move chown -R tomcat8:tomcat8 /var/lib/tomcat8 (250 MB)

For --no-install-recommends, I'd like to point out that **recommended but not mandatory packages are now skipped**, so this might require some testing to validate. 
As far as I could see, the xwiki instance built with this code is working perfectly, but there might be some hidden effects I could not detect. Also, I have no idea why you need libreoffice in the first place so I didn't know what to test ...

### CODE REVIEW

**Comment by vmassol** [comment]
Thanks!

> Also, I have no idea why you need libreoffice in the first place so I didn't know what to test ...

Because this container supports importing/viewing office documents.

File: xwiki-mysql-tomcat/Dockerfile [hunk 1]
```diff
@@ -32,12 +32,18 @@ ENV XWIKI_VERSION=8.4.4 \
 RUN echo "deb http://http.debian.net/debian jessie-backports main" > /etc/apt/sources.list.d/backports.list && \
   apt-get update && \
   apt-get -y upgrade && \
-  apt-get -y --force-yes install sudo nano unzip curl openjdk-8-jdk tomcat8 libreoffice
+  apt-get --no-install-recommends -y --force-yes install sudo nano unzip curl openjdk-8-jdk tomcat8 libreoffice && \
```
Comment (vmassol) [inline xwiki-mysql-tomcat/Dockerfile]: hmm this is strange, I had committed a change to have each package on a new line and I don't see it in this diff. See https://github.com/xwiki-contrib/docker-xwiki/blob/master/xwiki-mysql-tomcat/Dockerfile#L35

Any idea?

File: xwiki-mysql-tomcat/Dockerfile [hunk 2]
```diff
@@ -32,12 +32,18 @@ ENV XWIKI_VERSION=8.4.4 \
...
+  rm -rf /var/lib/apt/lists/*
```
Comment (vmassol) [inline xwiki-mysql-tomcat/Dockerfile]: BTW at some time I was doing a "apt-get clean" but apparently it got lost. Maybe it's nicer than the rm (are they doing the same thing)?

**Comment by zwindler** [comment]
For the "each package on a new line" comment, you commited it yesterday and
I did the git clone before that so that's why

For "apt-get clean" you're right it's nicer but it doesn't gain as much
(1MB vs 25MB). But if you don't like it, it's not the most important gain.
It's just something I have seen on many Dockerfiles but not a "standard"

2017-01-20 10:39 GMT+01:00 Vincent Massol <notifications@github.com>:

**Comment by vmassol** [comment]
> For "apt-get clean" you're right it's nicer but it doesn't gain as much
(1MB vs 25MB). But if you don't like it, it's not the most important gain.
It's just something I have seen on many Dockerfiles but not a "standard"

No I like it. So if we do the rm we don't need the "apt-get clean" right?

**Comment by zwindler** [comment]
You're correct, apt-get clean is not needed in this case

2017-01-20 10:44 GMT+01:00 Vincent Massol <notifications@github.com>:

**Comment by vmassol** [comment]
@zwindler can I merge your PR or are you sending a new one?

**Comment by zwindler** [comment]
Can you do the merge ?

I'm not yet confortable with git, this is my first pull request and I'm a
bit lost with merge and PRs and such.

Or tell me what you need me to do

2017-01-20 10:50 GMT+01:00 Vincent Massol <notifications@github.com>:

**Comment by vmassol** [comment]
Done, thanks!

**Pull request merged**